src/main/java/net/fabricmc/loom/configuration/ide/RunConfig.java

```diff
@@ -149,35 +149,92 @@ private static void populate(Project project, LoomGradleExtension extension, Run
 		}
 	}
 
-	public static RunConfig clientRunConfig(Project project) {
-		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
-
-		RunConfig ideaClient = new RunConfig();
-		ideaClient.configName = "Minecraft Client";
-		ideaClient.programArgs = "";
-		populate(project, extension, ideaClient, "client");
-		ideaClient.vmArgs += getOSClientJVMArgs();
-		ideaClient.vmArgs += " -Dfabric.dli.main=" + getMainClass("client", extension);
+	// Turns camelCase/PascalCase into Capital Case
+	// caseConversionExample -> Case Conversion Example
+	private static String capitalizeCamelCaseName(String name) {
+		if (name.length() == 0) {
+			return "";
+		}
 
-		return ideaClient;
+		return name.substring(0, 1).toUpperCase() + name.substring(1).replaceAll("([^A-Z])([A-Z])", "$1 $2");
 	}
 
-	public static RunConfig serverRunConfig(Project project) {
+	public static RunConfig runConfig(Project project, RunConfigSettings settings) {
 		LoomGradleExtension extension = project.getExtensions().getByType(LoomGradleExtension.class);
+		String name = settings.getName();
+
+		String configName = settings.getConfigName();
+		String mode = settings.getMode();
+		SourceSet sourceSet = settings.getSource(project);
+
+		String defaultMain = settings.getDefaultMainClass();
+
+		if (defaultMain == null) {
+			if (name.equals("client")) {
+				defaultMain = Constants.Knot.KNOT_CLIENT;
+			} else if (name.equals("server")) {
+				defaultMain = Constants.Knot.KNOT_SERVER;
+			} else {
+				// throw new IllegalArgumentException("Run configuration '" + name + "' must specify 'defaultMainClass'");
+
+				// Can go for above exception, or just assume user wants to start client (because that one is used most often)
```

Comment:
What the comment says: if no default main class is specified it can throw an exception (build fail) or just fall back on `KnotClient` (assuming client is most often run).